src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -58,52 +74,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
-  /**
-   * Gets all the comments associated with a certain recipe, based on recipe ID. For now, returns
-   * all the comments flatly. (prototype)
-   */
-  private String getComments(String recipeID, HttpServletResponse response) {
-    Query query = DBUtils.comments(recipeID);
-    ApiFuture<QuerySnapshot> querySnapshotFuture = query.get();
-    ArrayList<Object> commentsList = new ArrayList<>();
-
-    QuerySnapshot querySnapshot = DBUtils.blockOnFuture(querySnapshotFuture);
-    if (querySnapshot == null) {
-      return null;
-    }
-
-    for (DocumentSnapshot document : querySnapshot.getDocuments()) {
-      commentsList.add(document.getData());
-    }
-    return gson.toJson(commentsList);
-  }
-
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String commentData =
         request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
     String recipeID = request.getParameter("recipeID"); // Parent recipe of comment.
 
     String token = request.getParameter("token");
-    FirebaseToken decodedToken = Auth.verifyIdToken(token);
-    if (decodedToken == null) {
-      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-      return;
-    }
 
     Comment newComment = gson.fromJson(commentData, Comment.class);
     if (newComment.content == null) {
       response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
       return;
     }
-    newComment.creatorId = decodedToken.getUid();
-    ApiFuture addCommentFuture = DBUtils.comments(recipeID).document().set(newComment);
-    DBUtils.blockOnFuture(addCommentFuture);
+    String uid = Auth.getUid(token, response);
+    if (uid == null) {
+      return;
+    }
+
+    newComment.creatorId = uid;
```

Comment:
Here might be a good place to initialize other Comment properties, like timestamp, creator ldap, and votes